federation/planner.go

```diff
@@ -116,6 +139,30 @@ func printSelections(selectionSet *graphql.SelectionSet) {
 	}
 }
 
+// maybeCustomService adds the input selection to either localSelections or
+// selectionByService if there is a serviceSelector in the planner and the user
+// has custom service setting for the selection. It returns a boolean to indicates
+// whether there is a custom service for the selection.
+func (e *Planner) maybeCustomService(
+	typeName, currentService string,
+	selection *graphql.Selection,
+	localSelections []*graphql.Selection,
+	selectionsByService map[string][]*graphql.Selection,
+) (bool, []*graphql.Selection, map[string][]*graphql.Selection) {
+	if e.serviceSelector == nil {
+		return false, localSelections, selectionsByService
+	}
+	customService := e.serviceSelector(typeName, selection.Name)
```

Comment:
Should the custom service ever be an empty string? Will we just ignore the field if it is an empty string? I think I would prefer to fail the planning step of executing a query

Reply:
If the customer service is an empty string, it means that we don't have a custom config to the {typeName, selection} pair, and the first return value will be false. At line 198, we will continue the normal service assignment.

Reply:
I think we want to enforce that if the field exists on multiple services, the service selections can't return an empty string, and instead should have an error.

Reply:
Talked offline, I added a check for the case when there are more than one service associates with the selection field without a serviceSelecor, the planner will return an error. Also updated the unit test.